### ISSUE TITLE
Validate argument types on client before starting upload

### ISSUE THREAD

**Issue by sfc-gh-jfreeberg** [Snowflake-Labs/snowflake-maven-gradle-plugins#40]
In my `pom.xml` I had the following:

```xml
...
<procedure>
  <name>hello_world_proc</name>
  <handler>org.example.procedure.App.run</handler>
  <args></args>
  <returns>long</returns> <!-- problem -->
</procedure>
...
```

And it turns out long was incorrect (which may be a separate issue because we really should be inferring the SQL types from the Java types) and it went through the upload process, the ultimately failed at the CREATE step:

```log
(base) jfreeberg@YKQ2Q2CMYG snowpark-java-template % mvn snowflake:deploy 
[INFO] Scanning for projects...
[INFO] 
[INFO] ----------------< com.snowflake:snowpark-java-template >----------------
[INFO] Building Java 11 Project Template for Snowflake 0.0.1
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- snowflake-maven-plugin:0.1.0:deploy (default-cli) @ snowpark-java-template ---
[INFO] Execute copy dependencies. Destination: /Users/jfreeberg/Documents/GitHub/snowpark-java-template/target/dependency
[INFO] com.snowflake:snowpark:jar:1.8.0 already exists in destination.
[INFO] org.scala-lang:scala-library:jar:2.12.11 already exists in destination.
[INFO] org.scala-lang:scala-compiler:jar:2.12.11 already exists in destination.
[INFO] org.scala-lang:scala-reflect:jar:2.12.11 already exists in destination.
[INFO] org.scala-lang.modules:scala-xml_2.12:jar:1.0.6 already exists in destination.
[INFO] commons-io:commons-io:jar:2.11.0 already exists in destination.
[INFO] javax.xml.bind:jaxb-api:jar:2.2.2 already exists in destination.
[INFO] javax.xml.stream:stax-api:jar:1.0-2 already exists in destination.
[INFO] javax.activation:activation:jar:1.1 already exists in destination.
[INFO] org.slf4j:slf4j-api:jar:1.7.32 already exists in destination.
[INFO] org.slf4j:slf4j-simple:jar:1.7.32 already exists in destination.
[INFO] commons-codec:commons-codec:jar:1.15 already exists in destination.
[INFO] net.snowflake:snowflake-jdbc:jar:3.13.28 already exists in destination.
[INFO] com.github.vertical-blank:sql-formatter:jar:1.0.2 already exists in destination.
[INFO] com.fasterxml.jackson.core:jackson-databind:jar:2.13.4.2 already exists in destination.
[INFO] com.fasterxml.jackson.core:jackson-core:jar:2.13.2 already exists in destination.
[INFO] com.fasterxml.jackson.core:jackson-annotations:jar:2.13.2 already exists in destination.
[INFO] Can't extract module name from scala-xml_2.12-1.0.6.jar: scala.xml.2.12: Invalid module name: '2' is not a Java identifier
[INFO] Mapped dependencies to stage paths: {scala-compiler-2.12.11.jar=org/scala-lang/scala-compiler/2.12.11, activation-1.1.jar=javax/activation/activation/1.1, slf4j-simple-1.7.32.jar=org/slf4j/slf4j-simple/1.7.32, jackson-core-2.13.2.jar=com/fasterxml/jackson/core/jackson-core/2.13.2, scala-library-2.12.11.jar=org/scala-lang/scala-library/2.12.11, commons-codec-1.15.jar=commons-codec/commons-codec/1.15, jackson-annotations-2.13.2.jar=com/fasterxml/jackson/core/jackson-annotations/2.13.2, slf4j-api-1.7.32.jar=org/slf4j/slf4j-api/1.7.32, jackson-databind-2.13.4.2.jar=com/fasterxml/jackson/core/jackson-databind/2.13.4.2, stax-api-1.0-2.jar=javax/xml/stream/stax-api/1.0-2, scala-reflect-2.12.11.jar=org/scala-lang/scala-reflect/2.12.11, commons-io-2.11.0.jar=commons-io/commons-io/2.11.0, sql-formatter-1.0.2.jar=com/github/vertical-blank/sql-formatter/1.0.2, snowflake-jdbc-3.13.28.jar=net/snowflake/snowflake-jdbc/3.13.28, snowpark-1.8.0.jar=com/snowflake/snowpark/1.8.0, scala-xml_2.12-1.0.6.jar=org/scala-lang/modules/scala-xml_2.12/1.0.6, jaxb-api-2.2.2.jar=javax/xml/bind/jaxb-api/2.2.2}
[INFO] set JDBC client memory limit to 10240
[INFO] Creating connection to snowflake at url: jdbc:snowflake://https://pm.snowflakecomputing.com:443
[INFO] Snowflake Session established!
[INFO] Creating stage STAGE_NAME if not exists
[INFO] Stage located or created!
[INFO] Uploading artifact JAR: /Users/jfreeberg/Documents/GitHub/snowpark-java-template/target/snowpark-java-template-0.0.1.jar
May 12, 2023 11:40:34 AM net.snowflake.client.jdbc.internal.amazonaws.util.Base64 <clinit>
WARNING: JAXB is unavailable. Will fallback to SDK implementation which may be less performant.If you are using Java 9+, you will need to include javax.xml.bind:jaxb-api as a dependency.
[INFO] Artifact JAR uploaded!
[INFO] Uploading dependency JARs from: /Users/jfreeberg/Documents/GitHub/snowpark-java-template/target/dependency
[INFO] Uploading scala-compiler-2.12.11.jar
[INFO] Uploading activation-1.1.jar
[INFO] Uploading slf4j-simple-1.7.32.jar
[INFO] Uploading jackson-core-2.13.2.jar
[INFO] Uploading scala-library-2.12.11.jar
[INFO] Uploading commons-codec-1.15.jar
[INFO] Uploading jackson-annotations-2.13.2.jar
[INFO] Uploading slf4j-api-1.7.32.jar
[INFO] Uploading jackson-databind-2.13.4.2.jar
[INFO] Uploading stax-api-1.0-2.jar
[INFO] Uploading scala-reflect-2.12.11.jar
[INFO] Uploading commons-io-2.11.0.jar
[INFO] Uploading sql-formatter-1.0.2.jar
[INFO] Uploading snowflake-jdbc-3.13.28.jar
[INFO] Uploading snowpark-1.8.0.jar
[INFO] Uploading scala-xml_2.12-1.0.6.jar
[INFO] Uploading jaxb-api-2.2.2.jar
[INFO] Dependency JARs uploaded!
[INFO] Running create function statement: 
[INFO] CREATE OR REPLACE function combineStrings (a string, b string)
RETURNS string
LANGUAGE java
HANDLER = 'org.example.udf.Function.combineStrings'
IMPORTS = ('@STAGE_NAME/libs/snowpark-java-template-0.0.1.jar', '@STAGE_NAME/org/scala-lang/scala-compiler/2.12.11/scala-compiler-2.12.11.jar', '@STAGE_NAME/javax/activation/activation/1.1/activation-1.1.jar', '@STAGE_NAME/org/slf4j/slf4j-simple/1.7.32/slf4j-simple-1.7.32.jar', '@STAGE_NAME/com/fasterxml/jackson/core/jackson-core/2.13.2/jackson-core-2.13.2.jar', '@STAGE_NAME/org/scala-lang/scala-library/2.12.11/scala-library-2.12.11.jar', '@STAGE_NAME/commons-codec/commons-codec/1.15/commons-codec-1.15.jar', '@STAGE_NAME/com/fasterxml/jackson/core/jackson-annotations/2.13.2/jackson-annotations-2.13.2.jar', '@STAGE_NAME/org/slf4j/slf4j-api/1.7.32/slf4j-api-1.7.32.jar', '@STAGE_NAME/com/fasterxml/jackson/core/jackson-databind/2.13.4.2/jackson-databind-2.13.4.2.jar', '@STAGE_NAME/javax/xml/stream/stax-api/1.0-2/stax-api-1.0-2.jar', '@STAGE_NAME/org/scala-lang/scala-reflect/2.12.11/scala-reflect-2.12.11.jar', '@STAGE_NAME/commons-io/commons-io/2.11.0/commons-io-2.11.0.jar', '@STAGE_NAME/com/github/vertical-blank/sql-formatter/1.0.2/sql-formatter-1.0.2.jar', '@STAGE_NAME/net/snowflake/snowflake-jdbc/3.13.28/snowflake-jdbc-3.13.28.jar', '@STAGE_NAME/com/snowflake/snowpark/1.8.0/snowpark-1.8.0.jar', '@STAGE_NAME/org/scala-lang/modules/scala-xml_2.12/1.0.6/scala-xml_2.12-1.0.6.jar', '@STAGE_NAME/javax/xml/bind/jaxb-api/2.2.2/jaxb-api-2.2.2.jar');
[INFO] Running create function statement: 
[INFO] CREATE OR REPLACE procedure hello_world_proc ()
RETURNS long
LANGUAGE java
PACKAGES = ('com.snowflake:snowpark:latest')
HANDLER = 'org.example.procedure.App.run'
IMPORTS = ('@STAGE_NAME/libs/snowpark-java-template-0.0.1.jar', '@STAGE_NAME/org/scala-lang/scala-compiler/2.12.11/scala-compiler-2.12.11.jar', '@STAGE_NAME/javax/activation/activation/1.1/activation-1.1.jar', '@STAGE_NAME/org/slf4j/slf4j-simple/1.7.32/slf4j-simple-1.7.32.jar', '@STAGE_NAME/com/fasterxml/jackson/core/jackson-core/2.13.2/jackson-core-2.13.2.jar', '@STAGE_NAME/org/scala-lang/scala-library/2.12.11/scala-library-2.12.11.jar', '@STAGE_NAME/commons-codec/commons-codec/1.15/commons-codec-1.15.jar', '@STAGE_NAME/com/fasterxml/jackson/core/jackson-annotations/2.13.2/jackson-annotations-2.13.2.jar', '@STAGE_NAME/org/slf4j/slf4j-api/1.7.32/slf4j-api-1.7.32.jar', '@STAGE_NAME/com/fasterxml/jackson/core/jackson-databind/2.13.4.2/jackson-databind-2.13.4.2.jar', '@STAGE_NAME/javax/xml/stream/stax-api/1.0-2/stax-api-1.0-2.jar', '@STAGE_NAME/org/scala-lang/scala-reflect/2.12.11/scala-reflect-2.12.11.jar', '@STAGE_NAME/commons-io/commons-io/2.11.0/commons-io-2.11.0.jar', '@STAGE_NAME/com/github/vertical-blank/sql-formatter/1.0.2/sql-formatter-1.0.2.jar', '@STAGE_NAME/net/snowflake/snowflake-jdbc/3.13.28/snowflake-jdbc-3.13.28.jar', '@STAGE_NAME/com/snowflake/snowpark/1.8.0/snowpark-1.8.0.jar', '@STAGE_NAME/org/scala-lang/modules/scala-xml_2.12/1.0.6/scala-xml_2.12-1.0.6.jar', '@STAGE_NAME/javax/xml/bind/jaxb-api/2.2.2/jaxb-api-2.2.2.jar');
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  01:13 min
[INFO] Finished at: 2023-05-12T11:41:41-07:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal com.snowflake:snowflake-maven-plugin:0.1.0:deploy (default-cli) on project snowpark-java-template: Error creating function or procedure.: SQL compilation error:
[ERROR] Unsupported data type 'LONG'.
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
```  

The upload process can take a while (which is a separate area we can improve) so we should try to catch these unsupported argument types before we start the JAR upload.
